src/Application/DTO/RefundDto.php

```diff
@@ -18,6 +18,9 @@ public function setAmount(int $amount): self {
         if ($amount < 0) {
             throw new InvalidArgumentException("Refund amount cannot be negative.");
         }
+        if ($amount === 0) {
+            $amount = null;
+        }
```

Comment:
We shouldn't allow 0 as a valid amount. The amount is not mandatory, so if it's set, it must be valid.